src/features/form/validation/singleField/singleFieldValidationSagas.ts

```diff
@@ -35,14 +39,19 @@ export function* runSingleFieldValidationSaga({
     };
 
     try {
+      const layouts = yield select(selectLayoutsState);
+      const textResources = yield select(selectTextResources);
       const serverValidation: IValidationIssue[] = yield call(get, url, options);
-      const mappedValidations = mapDataElementValidationToRedux(
-        serverValidation,
-        state.formLayout.layouts || {},
-        state.textResources.resources,
-      );
 
-      const validations = mergeValidationObjects(state.formValidations.validations, mappedValidations);
+      const mappedValidations = mapDataElementValidationToRedux(serverValidation, layouts || {}, textResources);
+      const validationsFromState = yield select(selectValidationsState);
+      const validations = mergeValidationObjects(validationsFromState, mappedValidations);
+
+      // Reject validation if field is hidden
+      const hiddenFields = yield select(selectHiddenFields);
+      if (componentId in hiddenFields) {
+        throw Error('Single field validation rejected as field is hidden for user');
```

Comment:
1. Not sure if the `in` keyword works like that.. `in` works on keys, not values - and an array-type in JavaScript has numeric keys. See more here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in#basic_usage
2. Also, throwing an error probably makes the whole saga reject with the thrown error, storing the error in the Redux state and causing an `unknown error` to show on-screen. A bit aggressive for an error like this, which I think should be ignored silently (no need to update validations for components that have been hidden).
3. Additionally, if I were to read this code now (without knowing about this specific bug), I'd probably think "hey, there's no need to run this GET request first, then check if the field is hidden after getting a response - we can optimize away the whole request by just checking if the field is hidden before the request". I think we should check if the field is hidden _before_ sending the request as well, and update the comment here to describe that we meant to reject validation if the field _has been hidden in the time after we sent the request_ as well.

All in all, I think this warrants a cypress test. We should document this behavior via an e2e test, so as to make sure it doesn't regress in the future.